### PR TITLE
Lightening pipeline End to End with Hydra

### PR DESCRIPTION
**Pull request by Luckygyana**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,4 +36,5 @@ Usage:
 ./get_backgroundnoise.py
 ./preprocess.py
 ./transforms.py
+./run.py
```
Comment (turian) [inline README.md:39]: Do we really have to name this command "run"? I mean that's about as vague as it gets

**Comment by turian** [comment]
Please run `black` on all python files with default config

File: configs/callbacks/default.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+model_checkpoint_on_auc:
+  _target_: pytorch_lightning.callbacks.ModelCheckpoint
+  monitor: "val_acc" # name of the logged metric which determines when model is improving
```
Comment (turian) [inline configs/callbacks/default.yaml:3]: Let's just do val_auc for model checkpoint and for early stopping, to keep things simple and consistent

File: configs/optimizer/adamw.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# @package _group_
+class_name: torch.optim.AdamW
+params:
+  lr: ${training.lr}
+#    decoder_lr: 0.01
+  weight_decay: 0.001
```
Comment (turian) [inline configs/optimizer/adamw.yaml:6]: Should this be a parameter too?

File: configs/optimizer/sgd.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# @package _group_
+class_name: torch.optim.SGD
+params:
+  lr: ${training.lr}
+  momentum: 0.9
+  weight_decay: 0.0005
```
Comment (turian) [inline configs/optimizer/sgd.yaml:6]: Should this be a parameter too?

File: configs/scheduler/cosine.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# @package _group_
+class_name: torch.optim.lr_scheduler.CosineAnnealingLR
+step: epoch
+monitor: ${training.metric}
+params:
+  T_max: 10
```
Comment (turian) [inline configs/scheduler/cosine.yaml:6]: Should this be a parameter too?

Comment (turian) [reply configs/scheduler/cosine.yaml:6]: Etc, for the other optimizers, which do you expect to play with?

File: configs/callbacks/default.yaml (hunk 2 above)
Comment (Luckygyana) [reply configs/callbacks/default.yaml:3]: Okay
